### PR TITLE
feat: initial setup

### PR DESCRIPTION
**Pull request by AnnaRybkina**
**_What is included in this PR:_**

1. Migrate https://github.com/fabric-ds/css/blob/next/src/utils/component-classes/index.js from Fabric
2. Fix build step
3. Support for automated semantic releases. 

Changes introduced for the release are based on [our internal guide on adding automatic releases to Warp packages](https://github.com/warp-ds/release-playground/wiki/Automate-releases-with-semantic-release):
trigger a release with semantic-release whenever a commit is pushed to alpha or master branch.
3.1 add support for commitizen
3.2 add a PR title validator to ensure correct commit history structure in case of squashing
automatically update alpha branch whenever a merge to master takes place
Co-authored by @magnuh and @BalbinaK

### CODE REVIEW

**Comment by AnnaRybkina** [comment]
[Automate releases of warp packages to NPM and EIK](https://trello.com/c/v1Ptqg8G/128-automate-releases-of-warp-packages-to-npm)

File: index.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    warning: 'border-yellow-100 bg-yellow-100',
+    error: 'border-red-100 bg-red-100',
+    disabled: 'border-bluegray-300 bg-bluegray-300',
+    sponsored: 'border-aqua-200 bg-aqua-200',
+    neutral: 'border-gray-100 bg-gray-100',
+    outlined: 'border-bluegray-300 bg-white',
+};
+
+export const slider = {
+    wrapper: 'f-slider relative w-full h-44 py-2',
```
Comment (magnuh) [inline index.js:32]: All of the `f-` prefixes should be `w-`, I guess? But not sure if we should do that now or save it for later? 🤷

Comment (pearofducks) [reply index.js:32]: Those classes might be able to 🔥 now that we have better utility class support. I think wait-and-see is fine here?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "type": "module",
+  "exports": {
+    "*": "./dist/index.js"
+  },
+  "files": [
+    "dist"
+  ],
+  "scripts": {
+    "commit": "cz",
+    "build": "esbuild --bundle index.js --outdir=dist && tsc",
```
Comment (pearofducks) [inline package.json]: What is esbuild doing for us here?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@warp-ds/css",
+  "repository": "git@github.com:warp-ds/component-classes.git",
+  "version": "0.0.1",
+  "description": "WARP component classes",
+  "main": "dist/index.js",
+  "type": "module",
+  "exports": {
+    "*": "./dist/index.js"
```
Comment (pearofducks) [inline package.json]: I don't think this is the exports declaration we want?

Not sure I see a need for a `dist` when all we're doing is exporting a bunch of objects from a file. 🤷

File: package.json (hunk 2 above)
Comment (AnnaRybkina) [reply package.json]: Building a bundle to dist. But maybe that step is unnecessary? Are we ok with only exporting the index.js? And corresponding types?

File: index.js (hunk 1 above)
Comment (AnnaRybkina) [reply index.js:32]: Yea, let's leave it for later. A separate PR is also better to isolate the changes for the prefixes, I **_think_**

File: package.json (hunk 3 above)
Comment (AnnaRybkina) [reply package.json]: True, the export of index.js should be enough

File: .releaserc.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "branches": [{ "name": "master" }, { "name": "alpha", "prerelease": true }],
```
Comment (magnuh) [inline .releaserc.json:2]: Is it `master` or `main`? 😬😄

Comment (AnnaRybkina) [reply .releaserc.json:2]: master for this repo

Comment (pearofducks) [reply .releaserc.json:2]: `master` should be fine - it's the same setup as other projects

Comment (AnnaRybkina) [reply .releaserc.json:2]: I'm happy to change it when we have one way to go.

File: package.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
   "main": "dist/index.js",
   "type": "module",
   "exports": {
-    "*": "./dist/index.js"
+    "*": "./index.js"
```
Comment (pearofducks) [inline package.json]: I think `.` might be a better export target here?

File: package.json (hunk 3 above)
Comment (AnnaRybkina) [reply package.json]: Does this change make more sense https://github.com/warp-ds/component-classes/pull/2/commits/3466a9d0a29adab982eb9eed9c1ec8bd6cf15b0a

**Comment by pearofducks** [comment]
The `build` command and `esbuild` dep can be changed/removed - then this looks good!

**Comment by AnnaRybkina** [comment]
> The `build` command and `esbuild` dep can be changed/removed - then this looks good!

With [this change](https://github.com/warp-ds/component-classes/pull/2/commits/8095ae8fa4db0969c857a66e89f53d6555351528) we will release index.js to npm as is.

File: .releaserc.json (hunk 4 above)
Comment (magnuh) [reply .releaserc.json:2]: @warp-ds/warp-core-team I think we have a good mix of `main` and `master` as default branch in @warp-ds now. 🙈  What should we go with? (My vote is on `main`)

**Comment by magnuh** [comment]
Missing the LICENSE file? 🙂

**Review by BalbinaK — APPROVED** [review]
Looks great! 
Edit: didn't notice but yeah, the LICENSE file was missing 😄

**Review by pearofducks — APPROVED** [review]

**Comment by magnuh** [comment]
> didn't notice but yeah, the LICENSE file was missing 😄

I think it was in an initial (sneaky) commit to alpha. 😄

**Comment by AnnaRybkina** [comment]
> > didn't notice but yeah, the LICENSE file was missing 😄
> 
> I think it was in an initial (sneaky) commit to alpha. 😄

It was actually an initial commit to master https://github.com/warp-ds/component-classes/tree/master. I could choose it when I created the repo

File: .releaserc.json (hunk 4 above)
Comment (BalbinaK) [reply .releaserc.json:2]: I vote `main`, too. It's just one 1 syllable so I like it better 😄 Also: Warp = one syllable, vs Fabric = 2 syllables, sooooo
Old: Fabric-master
New: Warp-main
😄

**Comment by pearofducks** [comment]
That license file isn't valid - line 189 needs fixed

**Comment by AnnaRybkina** [comment]
> That license file isn't valid - line 189 needs fixed

Great catch! Adjusted

**Pull request merged**